app/src/main/kotlin/com/wire/android/ui/home/HomeTopBar.kt

```diff
@@ -37,31 +45,50 @@ fun HomeTopBar(
     viewModel: HomeViewModel
 ) {
     val titleText = stringResource(id = title ?: R.string.conversations_screen_title)
+    val scrollDownState = viewModel.scrollDown.observeAsState()
+    val firstLineElevation = if (!isSearchable || scrollDownState.value == true) dimensions().topBarElevationHeight else 0.dp
 
-    Column(
-        Modifier
-            .fillMaxWidth()
-            .background(MaterialTheme.colorScheme.background)
+    Box(
+        Modifier.background(Color.Transparent)
     ) {
-        CenterAlignedTopAppBar(
-            navigationIcon = { ToolbarIconBtn(scope, drawerState) },
-            title = { Text(text = titleText, style = MaterialTheme.wireTypography.title01) },
-            actions = {
-                UserProfileAvatar(avatarUrl = "", status = UserStatus.AVAILABLE) {
-                    scope.launch { viewModel.navigateToUserProfile() }
-                }
-            },
-            colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
-                containerColor = MaterialTheme.colorScheme.background,
-                titleContentColor = MaterialTheme.colorScheme.onBackground,
-                actionIconContentColor = MaterialTheme.colorScheme.onBackground,
-                navigationIconContentColor = MaterialTheme.colorScheme.onBackground
-            )
-        )
-
         if (isSearchable) {
-            SearchBarUI(placeholderText = stringResource(R.string.search_bar_hint, titleText.lowercase()))
+            val searchFieldFullHeightPx = LocalDensity.current.run {
+                (dimensions().topBarSearchFieldHeight + dimensions().topBarElevationHeight).toPx()
+            }
+            val searchFieldPosition by animateFloatAsState(if (scrollDownState.value == true) -searchFieldFullHeightPx else 0f)
+
+            Surface(
+                modifier = Modifier
+                    .padding(top = dimensions().smallTopBarHeight)
+                    .height(dimensions().topBarSearchFieldHeight)
+                    .graphicsLayer { translationY = searchFieldPosition },
+                shadowElevation = dimensions().topBarElevationHeight
+            ) {
+                SearchBarUI(
+                    modifier = Modifier.background(MaterialTheme.colorScheme.background),
+                    placeholderText = stringResource(R.string.search_bar_hint, titleText.lowercase())
+                )
+            }
         }
+
+        Surface(shadowElevation = firstLineElevation) {
```

Comment:
I was thinking of creating something similar to `SurfaceBackgroundWrapper` - `TopBarElevationWrapper` with parameters `withElevation` and `content` - we could wrap every TopBar with this simple composable and keep the TopBar elevation across the whole app unified

what do you think about that?

Reply:
TobBars are different in terms of its content inside, but visually I believe they all should be consistent and I think that Wolfgang wants all TobBars to work the same way throughout the whole app, but that may be just my own interpretation of his words:

> If the user starts scrolling the list down, a little shadow on the bottom of the appbar header is shown. 
> If the list is scrolled all to the top, the shadow is hidden.

If we create custom wrapper then we can set the default value, so the usage will look more like just:
`ElevationWrapper { /*Content here*/ }`
😄